### DISCUSSION TITLE
CSS templates

### DISCUSSION THREAD

**Discussion by ghelle-fhh** [ldproxy/ldproxy#948] [Ideas]
Hello, 

We've been playing around with the Moustache templates as we would like to add our corporate identity to the web page.

It's possible to change the CSS via style='', but that requires augmenting individual tags within the HTML structure.

Idealy it would be great to have seperate files where CSS could be writen and defined, in addition to the Moustache templates.

Or have I overlooked the feature?

Many thanks!

**Comment by azahnen** [comment]
Hi,

what you can do currently is override the `htmlStart` template, the original can be found here: https://github.com/interactive-instruments/ldproxy/blob/master/ogcapi-stable/ogcapi-html/src/main/resources/de/ii/ogcapi/html/templates/htmlStart.mustache

There you can either include a css file that is hosted somewhere else, define an inline stylesheet, or include an additional custom template with an inline stylesheet, e.g. with `{{> style-custom}}`. If you need a more elaborate example, let me know.

Overriding `htmlStart` (or actually any non-empty template) is not ideal since it may change on ldproxy updates, which can lead to problems. There is already an issue (#1003) to introduce empty templates at useful extension points and to support publishing of global assets like css files or images, which is currently scheduled for `v3.5.0`.
